staging/src/k8s.io/kube-aggregator/pkg/controllers/openapi/aggregator/aggregator.go

```diff
@@ -218,16 +219,21 @@ func (s *specAggregator) AddUpdateAPIService(apiService *v1.APIService, handler
 	s.mutex.Lock()
```

Comment:
if an API Service changes from remote to local (service != nil → service == nil), will the short-circuit return above prevent cleanup from happening? should we call RemoveAPIService in that case?

just noticed this, it's pre-existing in 1.27, so probably best to handle separately in a follow up

Reply:
ack, will do that in a follow up

Reply:
thanks, link the issue here when you open it to close the loop